pkg/pci/parse.go

```diff
@@ -0,0 +1,41 @@
+package pci
+
+import (
+	"bufio"
+	"bytes"
+)
+
+func isHex(b byte) bool {
+	return ('a' <= b && b <= 'f') || ('A' <= b && b <= 'F') || ('0' <= b && b <= '9')
+}
+
+// scan searches for Vendor and Device lines from the input *bufio.Scanner based
+// on pci.ids format. Found Vendors and Devices are added to the input map.
+// This implimentation expects an input pci.ids to have comments, blank lines, 
+// sub-devices, and classes already removed.
```

Comment:
I am pretty sure this function will work correctly if those changes have not been made, right? That's a good thing, so you might want to point it out.

Reply:
Feeding this function the upstream pci.ids, it currently will skip comments, blank lines, and sub-devices. However, it will incorrectly parse sub-classes as a devices under the last vendor, ffff  Illegal Vendor ID. Pushing a commit to fix that by checking if line[3] is a hex value. It will be a space on the sub-class lines.